server/services/recommendedVersionsService.ts

```diff
@@ -0,0 +1,219 @@
+  }
+
+  async getRecommendedVersions(): Promise<recommendedVersions> {
+    const now = Date.now()
+    if (this.cache && this.cache.expiresAt > now) {
+      return this.cache.versions
+    }
+
+    const result: recommendedVersions = {
+      helm_dependencies: {},
+      gradle: {},
+      metadata: { source: 'none', fetchedAt: new Date(now).toISOString() },
+    }
+
+    // Only Strapi: hmpps-template-kotlin component
+    const fromStrapi = await this.fetchRecommendedVersionsFromStrapi()
+    logger.info(`[RecommendedVersions] Strapi result: ${JSON.stringify(fromStrapi)}`)
+    if (fromStrapi) {
+      result.helm_dependencies.generic_prometheus_alerts = fromStrapi.helm_generic_prometheus_alerts
+      result.helm_dependencies.generic_service = fromStrapi.helm_generic_service
+      result.gradle.hmpps_gradle_spring_boot = fromStrapi.gradle_hmpps_gradle_spring_boot
+      result.metadata.source = 'strapi'
+    }
+
+    // Mark partial if any key is missing
+    if (
+      !result.helm_dependencies.generic_prometheus_alerts ||
+      !result.helm_dependencies.generic_service ||
+      !result.gradle.hmpps_gradle_spring_boot
+    ) {
+      result.metadata.source = result.metadata.source === 'none' ? 'none' : 'partial'
+    }
+
+    this.cache = {
+      versions: result,
+      expiresAt: now + this.ttlMillis,
+    }
+
+    logger.debug(
+      `[RecommendedVersions] Source=${result.metadata.source}, helm(gpa=${
+        result.helm_dependencies.generic_prometheus_alerts || 'missing'
+      }, gs=${result.helm_dependencies.generic_service || 'missing'}), gradle(hgsb=${
+        result.gradle.hmpps_gradle_spring_boot || 'missing'
+      })`,
+    )
+
+    return result
+  }
+
+  private parseVersionValue(raw: unknown): string | undefined {
+    if (raw === null || raw === undefined) return undefined
+    if (typeof raw === 'string' || typeof raw === 'number') return String(raw)
+    if (typeof raw === 'object') {
+      const obj = raw as Record<string, unknown>
+      return (obj.ref as string) || (obj.version as string) || undefined
+    }
+    return undefined
+  }
+
+  private async fetchRecommendedVersionsFromStrapi(): Promise<{
+    helm_generic_prometheus_alerts?: string
+    helm_generic_service?: string
+    gradle_hmpps_gradle_spring_boot?: string
+  } | null> {
+    try {
+      const templateComponentName = config.recommendedVersions.componentName
+      if (!templateComponentName) {
+        logger.warn('[RecommendedVersions] Missing template component name in config.recommendedVersions.componentName')
+        return null
+      }
+
+      const componentNameVariants = Array.from(
+        new Set(
+          [
+            templateComponentName,
+            templateComponentName.replace(/_/g, '-'),
+            templateComponentName.replace(/-/g, '_'),
+          ].filter(Boolean),
+        ),
+      )
+
+      let component: unknown | null = null
+      for (const variantName of componentNameVariants) {
```

Comment:
Could refine it further by using `const component = componentNameVariants.find(variantName +> {....` maybe 🤔